### PR TITLE
Better access to test results (WIP)

### PR DESCRIPTION
**Pull request by malteos**
WIP regarding https://github.com/deepset-ai/FARM/issues/332

- Optional multi processing for test dataset (map results back to original IDs)
- Optional evaluation after training
- Reports as dict (not str) and include logits

(I didn't look into unit tests so it might crash something at the current status)

### CODE REVIEW

**Comment by Timoeller** [comment]
Hey @malteos  thanks for the PR. Great to see people tweaking FARM! :dancers: 

**My general remark**: 
We would like to keep the Evaluator efficient and only produce eval metrics. Collecting all logits e.g. for the SQuAD dev set, would be quite an overhead.
If you want to output more stuff (like logits), we should bake this into the Inferencer (e.g. [here](https://github.com/deepset-ai/FARM/blob/master/farm/infer.py#L467)) and make the collection of logits optional in any case.
Btw. what do you need the logits for? Maybe there is a different way to work with them on your use case (in a PredictionHead)...

**Individual points:**
- Optional evaluation after training -> we have added this feature very recently in #335 but for your use case I believe it is sufficient to not load the test evaluator by setting test_filename=None in the Processor. (We should add documentation about this feature though.)
- Reports as dict (not str). That is a good idea, so people can access the results more individually. I think your intention was to access the logits though - right?

**Comment by Timoeller** [comment]
Hey @malteos I wanted to follow up on this PR.

I think especially the question: "what do you need the logits for?" could be interesting to work on together. Maybe there is a different way to work with the logits on your use case.

If we go forward with returning logits, we should put this feature into the inferencer and make it optional, since it will unnecessarily slow down most other use cases.

**Comment by malteos** [comment]
Sorry for not coming back to you. Other things are keeping me busy these days. 

I close this for now and may reopen it as soon as I continue working on it.
